src/cregistry/entry.c

```diff
@@ -517,16 +517,45 @@ int reg_entry_installed(reg_registry* reg, char* name, reg_entry*** entries,
  *
  * @param [in] reg     registry to search in
  * @param [in] path    path of the file to check ownership of
+ * @param [in] cs      false if check should be performed case-insensitive,
+ *                     true otherwise
  * @param [out] entry  the owner, or NULL if no active port owns the file
  * @param [out] errPtr on error, a description of the error that occurred
  * @return             true if success; false if failure
  */
-int reg_entry_owner(reg_registry* reg, char* path, reg_entry** entry,
+int reg_entry_owner(reg_registry* reg, char* path, int cs, reg_entry** entry,
         reg_error* errPtr) {
     int result = 0;
     sqlite3_stmt* stmt = NULL;
-    char* query = "SELECT id FROM registry.files WHERE actual_path=? AND active";
     int lower_bound = 0;
+    const char* base_query_start = "SELECT id FROM registry.files WHERE actual_path";
+    const char* base_query_mid = "?";
+    const char* base_query_end = " AND active";
+    const char* cis_operator = " LIKE ";
+    const char* cs_operator = "=";
+    const char* extend_query = " COLLATE NOCASE";
```

Comment:
Oh, apropos `LIKE`, do we actually want to use it? What if a path actually contains characters like `%` and `_`? Won't that lead to "interesting behavior"?

Given that both `COLLATE NOCASE` and `LIKE` perform a case-insensitive check for ASCII characters only, there isn't a lot of functional difference. If the ICU extension is available and loaded, `LIKE` can also lowercase non-ASCII characters, but I doubt Apple would build sqlite3 in that way. And since we don't have any control about the system sqlite3, it's probably better to assume that both `COLLATE NOCASE` and `LIKE` will only mangle pure ASCII characters.